aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,947 @@
+            // This is the collection that contains certificate tokens, which will allow users to exchange to an NFT in the destination collection after reveal time.
+            let source_collection_name = collection_config.collection_name;
+            string::append_utf8(&mut source_collection_name, b" - source collection");
+            create_collection(&resource_signer, source_collection_name, collection_config.collection_description, utf8(b""), collection_config.collection_maximum, collection_config.collection_mutate_config);
+
+            let source_token_data_id = create_tokendata(
+                &resource_signer,
+                source_collection_name,
+                // token name
+                utf8(b""),
+                // token description
+                utf8(b""),
+                // token maximum is equal to the number of tokens in the destination collection
+                vector::length(&collection_config.token_uri),
+                // token uri
+                utf8(b""),
+                signer::address_of(&resource_signer),
+                1,
+                0,
+                create_token_mutability_config(
+                    &vector<bool>[ false, false, false, false, true ]
+                ),
+                vector<String>[string::utf8(BURNABLE_BY_OWNER)],
+                vector<vector<u8>>[bcs::to_bytes<bool>(&true)],
+                vector<String>[string::utf8(b"bool")],
+            );
+
+            move_to(&resource_signer, SourceToken { token_data_id: source_token_data_id });
+        };
+
+        // if this is the whitelist minting time
+        if (now < whitelist_mint_config.whitelist_minting_end_time) {
+            let claimer_addr = signer::address_of(nft_claimer);
+            assert!(bucket_table::contains(&whitelist_mint_config.whitelisted_address, &claimer_addr), error::permission_denied(EACCOUNT_NOT_WHITELISTED));
+            let remaining_mint_allowed = bucket_table::borrow_mut(&mut whitelist_mint_config.whitelisted_address, claimer_addr);
+            assert!(*remaining_mint_allowed >= amount, error::invalid_argument(EAMOUNT_EXCEEDS_MINTS_ALLOWED));
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, whitelist_mint_config.whitelist_mint_price);
+                amount = amount - 1;
+                *remaining_mint_allowed = *remaining_mint_allowed - 1;
+            };
+        } else {
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, public_mint_config.public_mint_price);
+                amount = amount - 1;
+            };
+        };
+    }
+
+    // Exchange a source certificate token to a destination token. This function will burn the source certificate
+    // and put a destination token in the nft_claimer's TokenStore.
+    entry fun exchange(nft_claimer: &signer, token_id: TokenId) acquires NFTMintConfig, CollectionConfig, RevealConfig {
```

Comment:
TokenId cannot be entry argument

Reply:
updated to take `property_version` as an argument